api/base/serializers.py

```diff
@@ -82,15 +81,14 @@ def to_representation(self, value):
         """
         url = super(JSONAPIHyperlinkedIdentityField, self).to_representation(value)
 
-        if self.meta:
-            meta = {}
-            for key in self.meta:
-                meta[key] = _rapply(self.meta[key], _url_val, obj=value, serializer=self.parent)
-            self.meta = meta
-
-            return {'links': {self.link_type: {'href': url, 'meta': self.meta}}}
-        else:
-            return {'links': {self.link_type: url}}
+        meta = {}
+        for key in self.meta or {}:
+            if key == 'count':
+                show_related_counts = self.context['request'].query_params.get('related_counts', False)
+                if utils.is_falsy(show_related_counts):
```

Comment:
Probably should check for truthiness instead, since false is the default case. As it stands, if you were to, say, GET `http://localhost:8000/v2/nodes/?related_counts=fish`, you would get the meta information included.